src/outcome_spaces/sequential_pair_distances.jl

```diff
@@ -0,0 +1,104 @@
+is skipped and distances are computed directly on each state vector `xᵢ ∈ x`.
+
+## Description
+
+`SequentialPairDistances` does the following: 
+
+- Transforms the input timeseries `x` by first embedding it using embedding dimension
+    `m` and embedding lag `τ` (or skip this step if the input is already embedded).
+- Computes the distances `ds` between sequential pairs of points according to the given
+    `metric`.
+- Divides the interval `[minimum(ds), maximum(ds)]` into `n` equal-size bins by using 
+    [`RectangularBinEncoding`](@ref), then maps the distances onto these bins.
+
+## Outcome space
+
+The outcome space `Ω` for `SequentialPairDistances` are the bins onto which the 
+pairwise distances are mapped, encoded as the integers `1:n`. If you need the actual
+bin coordinates, these can be recovered with [`decode`](@ref) (see example below).
+
+## Implements
+
+- [`codify`](@ref). Note that the input `x` is ignored when calling `codify`, because
+    the input data is already handled when constructing a `SequentialPairDistances`.
+
+## Examples
+
+The outcome bins can be retrieved as follows.
+
+```julia
+using ComplexityMeasures
+x = rand(100)
+o = SequentialPairDistances(x)
+cts, outs = counts_and_outcomes(o, x)
+```
+
+Computing the "distribution entropy" with `n = 3` bins for the distance histogram:
+
+```julia
+using ComplexityMeasures
+x = rand(1000000)
+o = SequentialPairDistances(x, n = 3, metric = Chebyshev()) # metric from original paper
+h = information(Shannon(base = 2), o, x)
+```
+"""
+struct SequentialPairDistances{T, I<:Integer, M, DM, D, E} <: CountBasedOutcomeSpace
+    τ::T # ignored if input is `AbstractStateSpaceSet`
+    m::M 
+    n::I
+    metric::DM
+    dists::D
+    encoding::E
+end
+
+function SequentialPairDistances(x::AbstractVector; 
+        n = 3, m = 3, τ = 1, metric = Chebyshev())
+    x_embed = embed(x, m, τ)
+    dists = [metric(x_embed[i], x_embed[i + 1]) for i in 1:(length(x_embed) - 1)]
+    mindist, maxdist = minimum(dists), maximum(dists)
+    encoding = PairDistanceEncoding(mindist, maxdist; n, metric)
+    return SequentialPairDistances(τ, m, n, metric, dists, encoding)
+end
+
+function SequentialPairDistances(x::AbstractStateSpaceSet{m}; 
+        n = 3, metric = Chebyshev()) where m
+    dists = [metric(x[i], x[i + 1]) for i in 1:(length(x) - 1)]
+    mindist, maxdist = minimum(dists), maximum(dists)
+    encoding = PairDistanceEncoding(mindist, maxdist; n, metric)
+    return SequentialPairDistances(nothing, m, n, metric, dists, encoding)
+end
+
+# ----------------------------------------------------------------
+# Pretty printing (see /core/pretty_printing.jl).
+# ----------------------------------------------------------------
+hidefields(::Type{<:SequentialPairDistances}) = [:dists]
+hidefields(::Type{<:SequentialPairDistances{Nothing}}) = [:dists, :τ]
+
+total_outcomes(est::SequentialPairDistances) = est.n
+outcome_space(est::SequentialPairDistances) = collect(1:est.n)
```

Comment:
```suggestion
outcome_space(est::SequentialPairDistances) = 1:est.n
```
Is there any reason to explicitly `collect` here? The range should enough. It is sorted and has all other properties we need.